Graphics/Slicer/Math/Contour.hs

```diff
@@ -219,17 +219,13 @@ intersections dstPoint contour srcPoint = saneIntersections l0 $ contourLineSegs
     saneIntersections l1 ls = mapMaybe saneIntersectionOf ls
       where
         saneIntersectionOf :: LineSeg -> Maybe Point2
-        saneIntersectionOf l2 = saneIntersection (lineIntersection l1 l2)
+        saneIntersectionOf l2 = saneIntersection (intersectsWith (Left l1) (Left l2))
           where
-            saneIntersection :: Intersection -> Maybe Point2
-            saneIntersection (IntersectsAt p2) = Just p2
-            saneIntersection NoIntersection = Nothing
-            saneIntersection Parallel = Nothing
-            saneIntersection AntiParallel = Nothing
-    -- FIXME: fix these cases. steal the code / algorithms from infillLineSegInside.
-    --          saneIntersection Collinear = Nothing
-    --          saneIntersection LColinear _ _ = Nothing
-    --          saneIntersection (HitStartPointL2 p2) = Just p2
-    --          saneIntersection (HitEndPointL2 p2) = Just p2
+            saneIntersection :: Either Intersection PIntersection -> Maybe Point2
```

Comment:
Duplicate of `saneIntersection` above. Pull this out into a top level function.

Reply:
Probably not by the time i'm done with them. these should both expand to the "three at a time" form as in the other file, and need to handle colinear cases.

Reply:
I'd suggest to cross that bridge when you come to it, and to deduplicate in the meantime.